server/container/guice/common/src/main/java/org/apache/james/PeriodicalHealthChecks.java

```diff
@@ -46,7 +47,7 @@ public class PeriodicalHealthChecks implements Startable {
     private Disposable disposable;
 
     @Inject
-    PeriodicalHealthChecks(Set<HealthCheck> healthChecks, PeriodicalHealthChecksConfiguration configuration) {
+    PeriodicalHealthChecks(@Named("jamesHealthCheckWithAdditionalOnes") Set<HealthCheck> healthChecks, PeriodicalHealthChecksConfiguration configuration) {
```

Comment:
Maybe we can think of a better name for this injection.

How about... `@Named("resolved-checks") Set<HealthCheck>`

Reply:
Sure. BTW I have already written the integration test: `HealthCheckExtensionTest`.